view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js

```diff
@@ -712,6 +712,14 @@ define(
                         configuration.totalPriceLabel = configuration.configuration.merchantName;
                     }
                 }
+
+                // Show Merchant name for googlepay component
+                if (paymentMethod.methodIdentifier.includes('googlepay')) {
+                    if ('configuration' in configuration && !('merchantName' in configuration.configuration)) {
+                        configuration.configuration.merchantName = configuration.configuration.gatewayMerchantId;
```

Comment:
gatewayMerchantId will contain the merchants configured merchant name in a LIVE environment. So we can use this and not expose the merchant account code to the shopper